airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_table.py

```diff
@@ -292,3 +292,219 @@ def _read(cls,
+                                                                                          datafield=JsonPath("$.'data'[*]"),
+                                                                                          cursor_path=cursor_path)
+
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+    @mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+    @HttpMocker()
+    def test_incremental_without_initial_state_with_number_cursor(self, uuid_mock, mock_auth, http_mocker: HttpMocker) -> None:
+        _given_get_timezone(http_mocker)
+        _given_read_schema(http_mocker)
+        _given_table_catalog(http_mocker)
+        _given_table_with_primary_keys(http_mocker)
+
+        cursor_field = "ID"
+        id_index = 0
+        expected_cursor_value = 3
+        cursor_path = JsonPath(f"$.[{id_index}]")
+
+        http_mocker.post(
+            table_request()
+            .with_table(_TABLE)
+            .with_requestID(_REQUESTID)
+            .with_async()
+            .with_cursor_field(cursor_field)
+            .build(),
+            snowflake_response("async_response", FieldPath("statementStatusUrl"))
+            .with_handle(_HANDLE)
+            .build()
+        )
+
+        http_mocker.get(
+            table_request()
+            .with_handle(_HANDLE)
+            .build(is_get=True),
+            snowflake_response("response_get_table")
+            .with_record(self.get_record_builder_instance(cursor_path).with_cursor(1))
+            .with_record(self.get_record_builder_instance(cursor_path).with_cursor(2))
+            .with_record(self.get_record_builder_instance(cursor_path).with_cursor(3))
+            .build()
+        )
+
+        stream_builder = (SnowflakeStreamBuilder().with_name(self.stream_name)
+                          .with_cursor_field([cursor_field])
+                          .with_sync_mode(self.sync_mode))
+        catalog = CatalogBuilder().with_stream(stream_builder).build()
+
+        output = self._read(_config(), catalog)
+        most_recent_state = output.most_recent_state
+
+        assert len(output.records) == 3
+        assert most_recent_state.stream_descriptor == StreamDescriptor(name=self.stream_name)
+        assert most_recent_state.stream_state == AirbyteStateBlob(ID=expected_cursor_value)
+
+    @mock.patch("source_snowflake.streams.snowflake_parent_stream.uuid.uuid4", return_value=_REQUESTID)
+    @mock.patch("source_snowflake.source.SnowflakeJwtAuthenticator")
+    @HttpMocker()
+    def test_incremental_without_initial_state_with_date_cursor(self, uuid_mock, mock_auth, http_mocker: HttpMocker) -> None:
+        _given_get_timezone(http_mocker)
+        _given_read_schema(http_mocker)
+        _given_table_catalog(http_mocker)
+        _given_table_with_primary_keys(http_mocker)
+
+        cursor_field = "TEST_COLUMN_20"
+        id_index = 12
+        expected_cursor_value = "1970-01-04"
```

Comment:
Can we explain why `1970-01-04` is the expected value? Checking the records that are created [here](https://github.com/artefactory/airbyte/pull/37/files#diff-84ce19b13999dd0a0a5e7a0de9c7a4869deb78b1e04abb954161704c4b028b95R386-R388) does not explain clearly why we are passing from "3" to this date

Reply:
in snowflake api, for date type specifically, snowflake sends the number of days (positive or negative) from the date 1970-1-1 meaning that 3 is 1970-1-1 + 3 days gives 1970-1-4

Reply:
Cool! I'm fine with either adding a comment to explain that or use the code to document that. Something like:

```
most_recent_cursor = datetime(1970, 1, 4)
expected_cursor_value = most_recent_cursor.strftime("%Y-%m-%d")
```
and `(most_recent_cursor - datetime(1970, 1, 1)).days` to set the cursor value at line `377`

Reply:
Concerning documentation of this behavior, I prefer to document it at the place where the transformation happens to avoid duplication of explanation of behavior. I will put a comment to refer to the place where it is documented in the test. The transformation happens in this function source_snowflake.schema_builder.format_field. The next PR will concern a unittest about this with behavior documented for every case.